### ISSUE TITLE
Allow invoking Add all missing imports regardless of caret position

### ISSUE THREAD

**Issue by moniuch** [microsoft/vscode#140988]
TL;DR
I would like to be able to invoke Add all missing imports regardless of the caret placement.

Assume this TypeScript/RxJs code:

`refreshed$ = new Subject<void>()`

You can type this code and slow down a bit while typing the word `Subject` to give the auto import a chance to kick in and insert import for you.

Sometimes, however, you just want to dump the entire block of code out of your brain fast, and then - having the caret at the end of a line/block - invoke **Add all missing imports**, without having to jumping the caret back to a not imported symbol.

![image 2022-01-19 16-58-48](https://user-images.githubusercontent.com/681956/150167811-b9eeb9f5-c714-4815-8d24-0e3779b6ca75.png)

This command doesn't seem to be available from the palette while I am checking it now with the above code.

This would help with Dx.

**Comment by moniuch** [comment]
![image 2022-01-21 15-46-21](https://user-images.githubusercontent.com/681956/150547433-fb95e2f7-d2a0-47f9-9355-73a86d764921.png)

Also, in the above situation, `cmd-.` only offers individual imports, one by one. For better DX and a streamlined flow, I would see **Add all missing imports** available from the context menu

(Unfortunately I am not able to take a screenshot that includes the context menu, because it disappears when I click to drag a region)

**Comment by moniuch** [comment]
I think this issue could be extended to other areas where the position of the caret could be more relaxed in order to figure out the assistance needed.

Another example that just popped in was offering help for `implements`
If I type the following code and end up with caret in the middle of the empty class:

![image 2022-02-03 14-47-34](https://user-images.githubusercontent.com/681956/152355114-50ac0322-66e6-45ea-9116-b48515d382b4.png)

then I shouldn't really need to move back to the class name in order to `cmd-.` and be offered `Implement interface 'AsyncValidator'`. The caret anywhere within the class should be enough for that.

_I find it somewhat related to the OP, but if you prefer to have a separate ticket for that, please let me know._

**Comment by mjbvz** [comment]
This is already possible using the `source.addMissingImports` source actions. This can be trigged using the `Source Action...` command in js and ts files

For example, this setting applies the action on save:

```
    "editor.codeActionsOnSave": {
        "source.addMissingImports": true
    }
```

Improved support for edge cases is tracked by https://github.com/microsoft/TypeScript/issues/40921
